nixos/modules/services/video/wivrn.nix

```
@@ -0,0 +1,226 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  inherit (lib)
+    mkAliasOptionModule
+    mkIf
+    mkEnableOption
+    mkPackageOption
+    mkOption
+    mkDefault
+    optional
+    optionalString
+    optionalAttrs
+    isDerivation
+    recursiveUpdate
+    getExe
+    literalExpression
+    types
+    maintainers
+    ;
+  cfg = config.services.wivrn;
+  configFormat = pkgs.formats.json { };
+
+  # For the application option to work with systemd PATH, we find the store binary path of
+  # the package, concat all of the following strings, and then update the application attribute.
+  # Application can either be a package or a list that has a package as the first element.
+  applicationExists = builtins.hasAttr "application" cfg.config.json;
+  applicationListNotEmpty = (
+    if builtins.isList cfg.config.json.application then
+      (builtins.length cfg.config.json.application) != 0
+    else
+      true
+  );
+  applicationCheck = applicationExists && applicationListNotEmpty;
+
+  applicationBinary = (
+    if builtins.isList cfg.config.json.application then
+      builtins.head cfg.config.json.application
+    else
+      cfg.config.json.application
+  );
+  applicationStrings = builtins.tail cfg.config.json.application;
+
+  applicationPath = mkIf applicationCheck applicationBinary;
+
+  applicationConcat = (
+    if builtins.isList cfg.config.json.application then
+      builtins.concatStringsSep " " ([ (getExe applicationBinary) ] ++ applicationStrings)
+    else
+      (getExe applicationBinary)
+  );
+  applicationUpdate = recursiveUpdate cfg.config.json (
+    optionalAttrs applicationCheck { application = applicationConcat; }
+  );
+  configFile = configFormat.generate "config.json" applicationUpdate;
+in
+{
+  options = {
+    services.wivrn = {
+      enable = mkEnableOption "WiVRn, an OpenXR streaming application";
+
+      package = mkPackageOption pkgs "wivrn" { };
+
+      openFirewall = mkEnableOption "the default ports in the firewall for the WiVRn server";
+
+      defaultRuntime =
+        mkEnableOption ''
+          WiVRn Monado as the default OpenXR runtime on the system.
+          The config can be found at `/etc/xdg/openxr/1/active_runtime.json`.
+
+          Note that applications can bypass this option by setting an active
+          runtime in a writable XDG_CONFIG_DIRS location like `~/.config`
+        ''
+        // {
+          default = true;
+        };
+
+      autoStart = mkEnableOption "start the service by default";
```

Nit: Using mkEnableOption means that the description for this will be "Enable start the service by default", which is wrong. Override the description:

```suggestion
      autoStart = mkEnableOption "start the service by default" // {
        description = "Start the service by default";
      };
```

Or, use `mkOption` with `types.bool`.